content/blog/2018-08-14-where-to-bird.md

```diff
@@ -0,0 +1,365 @@
+``` r
+library("osmdata")
+```
+
+    ## Data (c) OpenStreetMap contributors, ODbL 1.0. http://www.openstreetmap.org/copyright
+
+``` r
+library("magrittr")
+(results <- opq(bbox = bbox) %>%
+  add_osm_feature(key = 'leisure', value = 'bird_hide') %>%
+  osmdata_sf ())
+```
+
+    ## Object of class 'osmdata' with:
+    ##                  $bbox : 47.6865,8.8753,47.8494,9.1177
+    ##         $overpass_call : The call submitted to the overpass API
+    ##             $timestamp : [ Thu 5 Jul 2018 08:06:35 ]
+    ##            $osm_points : 'sf' Simple Features Collection with 1 points
+    ##             $osm_lines : 'sf' Simple Features Collection with 0 linestrings
+    ##          $osm_polygons : 'sf' Simple Features Collection with 0 polygons
+    ##        $osm_multilines : 'sf' Simple Features Collection with 0 multilinestrings
+    ##     $osm_multipolygons : 'sf' Simple Features Collection with 0 multipolygons
+
+``` r
+results$osm_points
+```
+
+    ##              leisure            geometry
+    ## 5004940425 bird_hide 8.920901, 47.741569
+
+Yay, we now know where to find a bird hide not too far from the MPI!
+
+#### Visualizing our location and the bird hide
+
+So one could enter the coordinates of that bird hide in one’s favourite
+mapping software or app but to show you where the bird hide is we can
+actually step back and use
+[`osmplotr`](https://github.com/ropensci/osmplotr), another package
+contributed to rOpenSci by Mark Padgham!
+
+The way `osmplotr` works is letting you create a basemap, on which
+you’ll add different layers extracted from OpenStreetMap using
+`osmplotr::extract_osm_objects` or `osmdata` functions directly. Its
+strengths are therefore the use of open data, and the customization of
+what you’re using as background!
+
+Let’s create a basemap for our bounding box, and then add roads and
+buildings to it.
+
+``` r
+library("osmplotr")
+```
+
+    ## Data (c) OpenStreetMap contributors, ODbL 1.0. http://www.openstreetmap.org/copyright
+
+``` r
+bbox <- get_bbox(bbox)
+dat_B <- extract_osm_objects (key = 'building', bbox = bbox)
+dat_H <- extract_osm_objects (key = 'highway', bbox = bbox)
+
+map0 <- osm_basemap(bbox = bbox, bg = 'gray20') %>%
+  add_osm_objects (dat_B, col = 'gray40') %>%
+  add_osm_objects (dat_H, col = 'gray80')
+```
+
+``` r
+map0 %>%
+  add_axes() %>%
+  print_osm_map (filename = 'map_a1.png', width = 600,
+               units = 'px', dpi = 72)
+
+library("magrittr")
+magick::image_read('map_a1.png') %>%
+  magick::image_annotate("Map data © OpenStreetMap contributors",
+                         color = "white",
+                         boxcolor =  "black",
+                         size = 15,
+                         gravity = "south")
+```
+
+<img src="/img/blog-images/2018-08-14-where-to-bird/map1-print-1.png" width="600" />
+
+Quite pretty! The lakes can be seen because of the absence of roads and
+buildings on them.
+
+Now, let’s plot the bird hide and the MPI on them. Since we used
```

Comment:
Since you can't explain the background on every concept you're using, can you provide a set of links (maybe at the end of each post?) for people to read more about some concepts, like "sf" or working with spatial data in general?

Reply:
consider it done since it's done hehe.

Reply:
A really good link is the forthcoming book [Geocomputation in R](https://github.com/robinlovelace) by @robinlovelace, @nowosad, and @jannes-m

Reply:
the link to that book has been in the conclusion the whole time 😎